### PR TITLE
Add enum shape documentation

### PR DESCRIPTION
**Pull request by JordonPhillips**
This adds docs for the new enum shapes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/1.0/spec/core/traits.rst [hunk 1]
```diff
@@ -70,6 +70,77 @@ The ``trait`` trait is a structure that supports the following members:
...
 
+.. smithy-trait:: smithy.api#enumDefault
+.. _enumDefault-trait:
+
+---------------------
+``enumDefault`` trait
+---------------------
+
+Summary
+    Indicates that the targeted enum or intEnum member represents the default
```
Comment (kstich) [inline docs/source/1.0/spec/core/traits.rst:81]: Should `enum` and `intEnum` be refs?

File: docs/source/1.0/spec/core/traits.rst [hunk 2]
```diff
@@ -70,6 +70,77 @@ The ``trait`` trait is a structure that supports the following members:
...
+
+.. smithy-trait:: smithy.api#enumValue
+.. _enumValue-trait:
+
+-------------------
+``enumValue`` trait
+-------------------
+
+Summary
+    Defines the value of the targeted enum or intEnum member.
```
Comment (kstich) [inline docs/source/1.0/spec/core/traits.rst:117]: Should `enum` and `intEnum` be refs?

File: docs/source/1.0/spec/core/traits.rst [hunk 3]
```diff
@@ -70,6 +70,77 @@ The ``trait`` trait is a structure that supports the following members:
...
+Trait selector
+    ```:is(enum, intEnum) > member``
+Value type
+    string or integer
```
Comment (kstich) [inline docs/source/1.0/spec/core/traits.rst:121]: "string" and "integer" should be wrapped in ``.

File: docs/source/1.0/spec/core/traits.rst [hunk 4]
```diff
@@ -70,6 +70,77 @@ The ``trait`` trait is a structure that supports the following members:
...
+.. _enumValue-trait:
+
+-------------------
+``enumValue`` trait
+-------------------
+
+Summary
+    Defines the value of the targeted enum or intEnum member.
+Trait selector
+    ```:is(enum, intEnum) > member``
```
Comment (kstich) [inline docs/source/1.0/spec/core/traits.rst:119]: nit: extra opening `

**Review by kstich — CHANGES_REQUESTED** [review]

File: docs/source/1.0/spec/core/idl.rst [hunk 5]
```diff
@@ -705,6 +709,84 @@ The following example defines an ``integer`` shape with a :ref:`range-trait`:
...
+
+        @enumValue("HEART")
+        HEART
+
+        @enumValue("SPADE")
+        SPADE
+    }
+
+
+IntEnum shapes
```
Comment (kstich) [inline docs/source/1.0/spec/core/idl.rst:757]: nit: Should probably have its own heading, meaning the above would be the singular `idl-enum`.

Comment (JordonPhillips) [reply docs/source/1.0/spec/core/idl.rst:757]: This is its own heading, it's just not `ref`able. I'll add the ref target anyway

**Review by kstich — APPROVED** [review]

**Pull request merged**